src/library/trackmodel.h

```diff
@@ -50,7 +50,8 @@ class TrackModel {
     typedef int CapabilitiesFlags; /** Enables us to do ORing */
 
     enum SortColumnId {
-        SORTCOLUMN_INVALID = -1,
+        SORTCOLUMN_INVALID = -2,
```

Comment:
I am not particularly happy with this renumbering. Hopefully, -1 is not used literally anywhere in the existing code base like it is for invalid database ids. Using -1 for invalid ids is an omnipresent pattern in the code.

How can we be sure without turning `SortColumnId` into a type-safe `enum class`?

Reply:
I changed this on behalf of @Holzhaus , who introduced this enum in https://github.com/mixxxdj/mixxx/commit/5325677de850a5da25155e56d3e1a7ac10a1c0b3#diff-0d9324681e3f544722e2b0d3ee2834a83c25ce51988247b12865bb94ce2bc4f3
I guess he rated this enumeration change as safe, because it's new in 2.3 and therefore nobody could use it in his mappings.

Reply:
The alternative would be to use 0 for the current column and change all other enum values. Then you'd also need to check if any mappings in 2.3 or master already use this CO (at least the Roland DJ-505 does).

Reply:
...or use -2 as initially proposed? -1 is a very special value in the Mixxx code base and also in Qt for *invalid* values.

Reply:
To change all other enum values, use 0 for current column and -1 for invalid seems to be the cleanest state.
It would affect the Roland DJ-505 mapping and the Denon MC7000 mapping, which are part of the 2.3 branch. Therefore I could adjust them in this PR.
The risk regarding backward compatibility would be, that other users of 2.3 or master development builds used this new CO in their mappings too, but no user of a released Mixxx versions would be affected by this change.

Reply:
@Holzhaus Your opinion? I would support this decision for the sake of consistency, even though the actual values should not matter at all.

Reply:
Well, of course, the values matter once they have been defined!

We should add a note that these enum values are used literally by controller scripts and must never be changed! Unfortunately, the values of this enum are defined implicitly and not visible in the code.

Reply:
Both reordering or insertion of new enum variants is strictly forbidden! New variants must always be inserted between the last valid and before the terminating variant.

Reply:
< Unfortunately, the values of this enum are defined implicitly and not visible in the code.

Should I define all the values of this enum explicit, except NUM_SORTCOLUMNIDS of cause?

Reply:
That might help to communicate the intention and restrictions of this enum. We should also define SORTCOLUMN_ID_MIN/MAX, with NUM_SORTCOLUMNIDS = SORTCOLUMN_ID_COUNT = (SORTCOLUMN_ID_MAX - SORTCOLUMN_ID_MIN) + 1. Currently, the code implicitly assumes that 0 is the first and NUM_SORTCOLUMNIDS - 1 the last valid value, i.e. strictly 0-based.

Reply:
> Well, of course, the values matter once they have been defined!
> 
> We should add a note that these enum values are used literally by controller scripts and must never be changed! Unfortunately, the values of this enum are defined implicitly and not visible in the code.

Yes. In this special case it has never been part of a stable Mixxx release so I assume it's okay. I'd support redefining the column id such that 0 means the current column. It only affects two controller mappings:

```
$ git grep -l sort_column
res/controllers/Denon-MC7000-scripts.js
res/controllers/Roland_DJ-505-scripts.js
src/library/librarycontrol.cpp
src/widget/wtracktableview.cpp
```

Reply:
@JoergAtGithub When a decision is made here, can you please file a PR for the manual repo to update the CO documention: https://manual.mixxx.org/2.3/en/chapters/advanced_topics.html#control-[Library]-sort_column

Reply:
I searched through the code for NUM_SORTCOLUMNIDS, and found that the enum is not only used for the sorting feature, but also for import and export of different library formats. I found several arrays and loops like this:
`int m_columnIndexBySortColumnId[NUM_SORTCOLUMNIDS];`

`for (int i = 0; i < TrackModel::SortColumnId::NUM_SORTCOLUMNIDS; ++i) {`
`        m_columnIndexBySortColumnId[i] = -1;`
`}`
This all relies on the first ID of a real column to be zero.

To change this would make this code complicated and difficult to read. Therefore I sugest not to shift the definition of this enumeration.

Reply:
These loops are for looking up the actual column id for a given sort column ID. I think you need `m_columnIndexBySortColumnId[0] = -1` to indicate that SortColumnId 0 does not map to a special column.

Reply:
I changed the values of the SORTCOLUMIDs now as suggested.
I tested sort_column_toggle with all values from -1 to 30:
- I couldn't find a table view, where SORTCOLUMN_SAMPLERATE (value 29) is used
- Sorting with SORTCOLUMN_LOCATION (value 25) seems to be not implement - it does not work for this column. But I think, this is independent from this PR